### PR TITLE
fix(atsu): FOB information is reset when Refuel *LOAD pressed in MCDU mid-flight

### LINKED ISSUE #3502: [BUG] FOB information is reset when Refuel *LOAD pressed in MCDU mid-flight
**Mod Version**
Latest stable v0.5.2

**Describe the bug**
While in mid flight, pressing on the Refuel *LOAD button resets the FOB value of the aircraft.

I was performing a flight from RPLL -> RJAA and was navigating through my MCDU when I inadvertently pressed the Refuel *LOAD button. While I had already spent a significant amount of fuel at this point, FOB was reset to what it was at the beginning of my flight (27333 lbs). 

This Block Fuel value was taken from Simbrief, which I had loaded into the MCDU.

**To Reproduce**
1. Load a Simbrief flight plan to the MCDU
2. Take off and fly to your CRZ altitude or any point where there's a significant amount of fuel burn already
3. Press MCDU MENU -> ATSU -> AOC MENU -> PEFR/W&B screen
4. Press the button beside Refuel *LOAD

**Expected behavior**
FOB will reflect actual remaining fuel on board

**Actual behavior**
FOB resets to Block Fuel value set in Simbrief

**References**
ECAM lower display shows the fuel spent at 10830 lbs, but FOB value here and at the Upper ECAM display shows 25480 lbs, after about 10 minutes flight time since pressing the Refuel *LOAD button:
FOBBug

**Additional context**
The Block Fuel value comes from Simbrief.

### PR DESCRIPTION
**Pull request by erso44**
Fixes #3502 

## Summary of Changes
Aircraft can be fueled via ATSU only when ground control is ON.
The reason for this idea is that ground control is only ON when aircraft is on ground. When control is OFF fueling via ATSU is not working. Ground control is not activate while taxiing, flying etc..

Here comes another issue:
Right now ground control can be activated in flight. Someone needs to work on this.

If there is any problem regarding my solution please get in contact with me.

## Screenshots (if necessary)
not necessary

## References
not necessary

## Additional context
Please get in touch with me if there is any problem with my solution.

Discord username (if different from GitHub): erso44 or Ersin Karakilic

## Testing instructions
Aircraft on ground -> ATSU fueling works
Aircraft in flight -> ATSU fueling not working

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js [hunk 1]
```diff
@@ -157,10 +157,13 @@ class CDUAocOfpData {
             return mcdu.getDelayBasic();
         };
         mcdu.onLeftInput[4] = async () => {
-            if (currentBlockFuel) {
-                loadFuel(mcdu, updateView);
+            const isOnGround = SimVar.GetSimVarValue('L:A32NX_RCDR_GROUND_CONTROL_ON', 'bool');
```
Comment (sidnov) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js:160]: how about using the ```SIM ON GROUND``` simvar instead. That's the one used for the EFB as well.

**Review by sidnov — CHANGES_REQUESTED** [review]

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js (hunk 1 above)
Comment (erso44) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js]: rework done.

Comment (ExampleWasTaken) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js]: @sidnov @erso44 What about using `Simplane.getIsGrounded()`?

Comment (erso44) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js]: I am currently also adding GPS GROUND SPEED it should be enough.

Comment (ExampleWasTaken) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js]: > 
> 
> I am currently also adding GPS GROUND SPEED it should be enough.

My point is code consistency. The method I suggested is usually used to test if the aircraft is on ground.

Comment (erso44) [reply flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_OFPData.js]: Also before you decide to merge my PR I have to change the changelog which is wrong. I don't want to commit to much.

**Review by Benjozork — APPROVED** [review]

**Review by sidnov — APPROVED** [review]
Good to go 👍🏻, should be building now. In the meantime, please update the testing instructions/description as per the changes made

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #5105 
Tier of Testing: 1
Date: 23/06/2021

**Testing Process:**
1. Loaded C&D, tested ATSU refueling
2. Loaded on RW, tested ATSU refueling
3. Loaded into flight, tested ATSU refueling
**Negatives:** N/A
**Testing Results:** Passed/Not Passed

Conclusions: (Notes made, the positives in the PR, and anything extra you would like to mention - Delete Line if Nothing Extra is to be mentioned)
Media: 
C&D

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #5105 
Tier of Testing: 1
Date: 23/06/2021

**Testing Process:**
1. Loaded C&D, tested ATSU refueling (manual and simbrief import) -> no fueling
2. Test abort

**Negatives:**
- ATSU fueling was not possible even on the stand, so I had no comparison to continue. What did I wrong?

**Testing Results:** Not Passed

Media: 
C&D
image

**Comment by erso44** [comment]
I have fixed my error regarding GS and made some improvment.

Aircraft needs to be on ground.
GS must be less than 1 -> @St54Kevin said that it could be possible that GS increases on strong wind speeds. I tested with extrem windspeeds and ATSU refueling was working.
No engine is running (improvment)

**Comment by 2supersanjay** [comment]
### Quality Assurance Trainee Report

Object of testing: PR #5105 
Tier of Testing: 1
Date: 2/7/2021

### Testing Process: 

1. Loaded up cold and dark at the gate
2. Manually imputed block/taxi/trip fuel
3. Resetted the values for the fuel
4. Loaded up the OFP request in the ATSU
5. Started a new flight on the runway
6. Took off and cruised at 3000ft
7. Manually imputed block/taxi/trip fuel
8. Resetted the values for the fuel
9. Loaded up the OFP request in the ATSU
10. Landed at the same airport
11. Manually imputed block/taxi/trip fuel while the engines were running and the GS was at '0'
12. Resetted the values for the fuel
13. Loaded up the OFP request in the ATSU while the engines were running and the GS was at '0'

### Negatives: 

- One doubt I had was when I clicked refuel for the right scenarios. It will fill up the tanks but be 20KGs short after I refueled it. So, when refuelling the amount of fuel put into the plane isn't exact to what I put as the block fuel. 

### Testing Results: Passed

### Conclusions: 

- Everything works properly. If the GS is above '0' and if the engines are running, there is no way to use the ATSU to refuel the aircraft. Otherwise, if the engines are off and the GS is '0', then the aircraft will refuel properly. 

Media: https://youtu.be/fxQHGQsFQ1Y

**Comment by St54Kevin** [comment]
Have you considered working together with #5254 for this?

**Comment by erso44** [comment]
Why ?

**Comment by St54Kevin** [comment]
> Why ?

Because you are using some sort of "when on ground/airborne" logic here right? And #5254 replaces our old logic

**Comment by erso44** [comment]
> > Why ?
> 
> Because you are using some sort of "when on ground/airborne" logic here right? And #5254 replaces our old logic

because he started his ticket after mine and I was already finished ?? did you check the commit timestamp before starting to complain ?

**Comment by Benjozork** [comment]
What is the status on this

**Comment by erso44** [comment]
Looks like tests passed

**Comment by marcman86** [comment]
Quality Assurance Tester Report
Discord:marcman86#4907
Object of testing: #5105
Tier of Testing: 1
Date: 10/07/2021

Testing Process:
- Spawned C&D at EDDP
- Turned on bat. and ext. power
- Refueled using ATSU
- Start engines and try ATSU refueling > inhibited
- Took off and try ATSU refueling in air > inhibited

Negatives: -

Testing Results: Passed
Conclusions: Works. Would be nice if the MCDU textfield *LOAD be greyed out or invisible when refueling is inhibit.
Hope this helps :-)

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #5105 
Tier of Testing: 1
Date: 12/07/2021

**Testing Process:**
1. Loaded C&D with a FPLN from simbrief, powered up
2. Tested ATSU simbrief refueling and manual value input
3. Started ENG, tested ATSU refueling
4. Takeoff, tested ATSU refueling

**Negatives:** N/A
**Testing Results:** Passed

**Comment by TheSquirtingPantin** [comment]
Quality Assurance Tester

Discord: Remi#8871
Object of testing: #5105
Tier of Testing: 2
Date: 12/07/2021

Testing Process:
1) Flight KIAD-KORD
2) Attempted refuel during cruise using the ATSU page

Negatives: N/A

Conclusions: PASS. Refuel not possible during the airborne phase. 
Media: N/A

**Comment by FlightSimmer7700** [comment]
Quality Assurance Tester

Discord: FlightSimmer7700#3468
Object of testing: #5105 
Tier of Testing: 1
Date: 14/7/21

Testing Process: 
Loaded C&D and tried
Loaded on runway and tried
Loaded mid-air and tried
Loaded on approach and tried
Completed a full circle and checked it was only available with Ground Control on

Negatives: Not an issue but would be better if something was displayed in the scratchpad to say that it won't affect anything otherwise it looks bugged or like you didn't press it
Testing Results: Passed

Conclusions: LGTM as a function but a message on the scratchpad would probably help

**Comment by 2hwk** [comment]
@erso44 Thank you and congratulations for your first (officially merged) contribution to the project. Will be merging shortly.

**Pull request merged**